app/js/view/main_view.js

```diff
@@ -14,16 +18,27 @@ export default class MainView extends View {
           window.dispatchEvent(new CustomEvent('request-activity-finish'));
         }
       });
+    } else {
+      var uploadBtn = document.getElementById('upload-link');
```

Comment:
It's weird to bind to DOM external to the view like this. The view isn't actually constructing this part of the DOM, so it should be passed into the ctor like this:
```js
new MainView({uploadBtn: document.getElementById('upload-link')});
```
This is a bit nit-picky, so under the circumstances, I can let this one slide.